Resurrects Cassandra's DependencyStore

codefromthecrypt opened the pull request:
This resurrects Cassandra's DependencyStore, and notes the constraints
of the design. Most importantly, calls to `storeDependencies` will
overwrite values on a day granularity. This is different than anorm, for
example, which doesn't bucket by day.

See #597

codefromthecrypt commented:
rebased, but no other progress

eirslett commented:
Oh, sorry for the lack of response!
+1 LGTM!

codefromthecrypt commented:
To whom it may concern. I looked into resurrecting this. The problem is that there is very little test coverage to say what "should" happen. For example, the tests are trivial and rely mostly on whatever algebird is supposed to do. Best first step would be to backfill tests for the working implementation (anormdb and/or zipkin-hbase), then try to make a cassandra impl that passes those tests.

rtyler commented:
See also [Working Effectively With Legacy Code](http://objectmentor.com/resources/articles/WorkingEffectivelyWithLegacyCode.pdf) <_<

Commit:
Resurrects Cassandra's DependencyStore

This resurrects Cassandra's DependencyStore, and notes the constraints
of the design. Most importantly, calls to `storeDependencies` will
overwrite values on a day granularity. This is different than anorm, for
example, which doesn't bucket by day.

codefromthecrypt commented:
PTAL!

codefromthecrypt commented on zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraDependencyStore.scala:
```diff
@@ -0,0 +1,39 @@
+import scala.collection.JavaConverters._
+
+/**
+ * This implementation of DependencyStore assumes that the job aggregating dependencies
```
this thing was the reason it took me so long to finish this. eventually, I figured out why everything was flooring to a day!

codefromthecrypt commented:
merging so I can release, which could help progress https://github.com/openzipkin/zipkin-dependencies-hadoop
